scripts/generate-js-globals.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+const vm = require("vm");
+const fs = require("fs");
+const path = require("path");
+
+// Creates a list of globals from the JS environment (not the web), so that Window.js can alias them when necessary.
+// The generated list should match https://tc39.es/ecma262/#sec-global-object, to the extent V8 implements the spec.
+// We generate this at build time instead of runtime because we want to avoid the performance and memory overhead of
+// creating a new context when scripting is disabled in the JSDOM.
+
+const dest = path.resolve(__dirname, "../lib/jsdom/browser/js-globals.json");
+
+const context = vm.createContext();
+const globals = vm.runInContext("Object.getOwnPropertyNames(this)", context);
+
+// I guess VM contexts have a console, from V8? That isn't a JS global and JSDOM will install its own, so don't include
```

Comment:
The default console connects to the Inspector, if it's enabled.

Reply:
It's still surprising this would be injected into VM contexts, IMO

Reply:
VM contexts are treated by the Inspector similarly to iframes in HTML. That's why `debugger` statements in jsdom-rendered HTML files actually work, for otherwise `console.log` in the DevTools console wouldn't even work.